packages/core/src/storage/extract.ts

```diff
@@ -108,3 +149,23 @@ function getTypeMembers(typeDef: StructDefinition | EnumDefinition): TypeItem['m
     return typeDef.members.map(m => m.name);
   }
 }
+
+function getOriginContract(
+  basecontract: ContractDefinition,
+  astId: number | undefined,
+  deref: ASTDereferencer,
+): [VariableDeclaration | undefined, string] {
+  for (const id of basecontract.linearizedBaseContracts.reverse()) {
+    if (id === basecontract.id) {
+      continue;
+    }
+    const parentContract = deref(['ContractDefinition'], id);
+
+    const varDecl = parentContract.nodes.filter(n => n.id == astId)[0];
+    if (varDecl && isNodeType('VariableDeclaration', varDecl)) {
+      return [varDecl, parentContract.name];
+    }
+  }
+
+  return [undefined, ''];
```

Comment:
If we change the type as I suggested above, the explicit return statement is not necessary.